client.go

```diff
@@ -763,15 +768,11 @@ func (f *File) Read(b []byte) (int, error) {
 // WriteTo writes the file to w. The return value is the number of bytes
 // written. Any error encountered during the write is also returned.
 func (f *File) WriteTo(w io.Writer) (int64, error) {
-	fi, err := f.Stat()
-	if err != nil {
-		return 0, err
-	}
 	inFlight := 0
 	desiredInFlight := 1
 	offset := f.offset
 	writeOffset := offset
-	fileSize := uint64(fi.Size())
+	fileSize := uint64(f.size)
```

Comment:
Using this cached size, we run into a problem. Follow these steps, create a new file (`f.size == 0`), now write out 512 bytes to that file. Seek to the begining, and now, make a call to WriteTo. This will assume the file is zero-length.